### ISSUE TITLE
IndentationError: unindent does not match any outer indentation level

### ISSUE THREAD

**Issue by knjpollard** [googlecolab/colabtools#1904]
Bug report for Colab: http://colab.research.google.com/.

For questions about colab usage, please use [stackoverflow](https://stackoverflow.com/questions/tagged/google-colaboratory/).

- Describe the current behavior:
The error happens even though there is no indent problem. I've tried 4 spaces 8 spaces. The cell seems to get jacked when copying and pasting. Once it starts you can't get rid of it. 

- Describe the expected behavior:
No error should come. I mean indenting is easy you look at the vertical lines and all items should be lined up with those lines in 4 space increments. In my case they are. I have 100s of lines of code "without this issue" but this one cell, because I copied and pasted into it, has an issue.  Possibly bad chars getting into the cell, but i have tried pasting into notepad then back into the cell but no avail. 

The SAME CELL RUNS LOCALLY IN MY JUPYTERLAB IDE (caps are just to emphasize it's not the code). If it was it would NOT run in my local copy within Jupyterlab. 

- The web browser you are using (Chrome, Firefox, Safari, etc.):
chrome but it doesn't matter

- Link (not screenshot!) to a *minimal, public, self-contained* notebook that

Here's the code that breaks....notice it's just compiling a function. Nothing actually has to run here. 

# evaluate a model using k-fold cross-validation
def evaluate_model(dataX, dataY, n_folds=5):
	scores, histories = list(), list()
	# prepare cross validation with shuffle
	kfold = KFold(n_folds, shuffle=True)# fold_no =1    
	fold_no =1 # enumerate splits
	for train_ix, test_ix in kfold.split(dataX,dataY):
		# define model
		model = final_model_cv        
        # select rows for train and test
		ix_train, iy_train, ix_test, iy_test = dataX[train_ix], dataY[train_ix], dataX[test_ix], dataY[test_ix]
		# fit model
        # add callback for Tensorboard
        # set erbose = 0 to mute model training outputut
		final_cv_history = model.fit(ix_train, iy_train, epochs=epochs, batch_size=batch_size,validation_data=(ix_test, iy_test),verbose=1,callbacks=[check_point_callback])
    	# evaluate model
		_, cv_accuracy = model.evaluate(ix_test, iy_test, verbose=0)
		print('> %.3f' % (cv_accuracy * 100.0))
		# append scores
		scores.append(cv_accuracy)
		histories.append(final_cv_history)
        **fold_no = fold_no + 1**       
	return scores, histories

The part above in bold is which results in the "indent error," but it is in fact indented identically to the line above it. Now if i comment that line out, then the function compiles without error. IMPORTANT! The line got there by way of copy paste, so there is something with that. 

Bear in mind the code here of course doesn't paste with the indents!

**UPDATE**

I found the bug and I have a workaround. When you "cut and past code" from somewhere else and into the cell. It's not taking the spaces and tabs with it. And it doesn't allow you put spaces and tabs into it post copy and past, which is the bug. 

The workaround. Cut and past the actual blank space between the left boarder and the code line from any cell that is working properly and that puts the spaces and tabs back in front of the offending line.

So in short, the **problem is the cell is not able to deal with cut and past**, especially code you have pasted from the internet or a kindle book or several other places.
